.github/workflows/CI.yml

```
@@ -8,6 +8,11 @@ on:
     tags: "*"
   schedule:
     - cron: "0 2 * * *"  # Daily at 2 AM UTC (8 PM CST)
+concurrency:
+  # Skip intermediate builds: all builds except for builds on the `master` branch
+  # Cancel intermediate builds: only pull request builds
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref != 'refs/heads/master' || github.run_number }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
```

I can't think of a good reason not to cancel in progress jobs for branches and tags. Is there a scenario I'm not thinking of?
```suggestion
  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```